elastic/integrations
# lastpass: Add `data.pagesize` and `data.pageindex` to `detailed_shared_folder` datastream

## Proposed commit message
```
Add pagination to `detailed_shared_folder` data.

When `data.pagesize` and `data.pageindex` properties are omitted from the 
request body, by default the API returns all shared folders. Internally, LastPass does 
multiple backend calls which could result in `429 Too Many Requests`. 
To prevent this, add `data.pagesize` and `data.pageindex` to request body.

Other changes:
  - Update request `body.data: all` to `body.data.data: all` as per the API guide.
  - Update system tests to contain pagination request.
  - Add `assert.hit_count` to system test config.
```

Reference - https://support.lastpass.com/s/document-item?language=en_US&bundleId=lastpass&topicId=LastPass/api_shared_folder_detailed_data.html&_LANG=enus



## Checklist


## Author's Checklist


- [x] Check pagination using system tests

## How to test this PR locally


`elastic-package stack down && elastic-package build && elastic-package stack up -d -v && eval "$(elastic-package stack shellinit)" && elastic-package test system --data-streams=detailed_shared_folder --generate  -v --defer-cleanup 30m `
```
2024/05/24 16:58:38 DEBUG Policy revision assigned to the agent (ID: b0183e4b-ecd8-4ee9-8e73-bb43ce3ddcf2)...
2024/05/24 16:58:38 DEBUG checking for expected data in data stream (10m0s)...
2024/05/24 16:58:38 DEBUG found 0 hits in logs-lastpass.detailed_shared_folder-ep data stream
2024/05/24 16:58:50 DEBUG found 8 hits in logs-lastpass.detailed_shared_folder-ep data stream
2024/05/24 16:58:54 DEBUG found 8 hits in logs-lastpass.detailed_shared_folder-ep data stream
2024/05/24 16:58:54 DEBUG running command: /usr/local/bin/docker compose version --short
2024/05/24 16:58:54 DEBUG Determined Docker Compose version: 2.23.0-desktop.1
2024/05/24 16:58:54 DEBUG running command: /usr/local/bin/docker compose -f /Users/kcreddy/go/src/github.com/kcreddy/integrations/packages/lastpass/_dev/deploy/docker/docker-compose.yml -p elastic-package-service ps -a -q lastpass
2024/05/24 16:58:54 DEBUG output command: /usr/local/bin/docker inspect 033eff719d9a40a408c8fb4f638ffc7d1ed4a4d4afb67076158519f187b1e9ba
2024/05/24 16:58:55 DEBUG check whether or not synthetics is enabled (component template logs-lastpass.detailed_shared_folder@package)...
2024/05/24 16:58:55 DEBUG data stream logs-lastpass.detailed_shared_folder-ep has synthetics enabled: false
2024/05/24 16:58:55 DEBUG Package does not embed ECS mappings
2024/05/24 16:58:55 DEBUG assert hit count expected 8, observed 8
.......
--- Test results for package: lastpass - START ---
╭──────────┬────────────────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE  │ DATA STREAM            │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────┼────────────────────────┼───────────┼───────────┼────────┼───────────────┤
│ lastpass │ detailed_shared_folder │ system    │ default   │ PASS   │ 34.453457458s │
╰──────────┴────────────────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: lastpass - END   ---
Done
```

## Screenshots
